Dec2022Demo/LibForOct2022Demo.jl

```diff
@@ -0,0 +1,348 @@
+#=module Oct2022Demo
+
+export stratify, typed_stratify, simulate, calibrate, MakeReactionSystem, generate_data, get_infected_states, get_susceptible_states,
+    get_recovered_states, optimise_p, full_train, obs_SIRD, makeLossFunction, plot_obs_w_ests, full_train_rand
+ =#
+
+using AlgebraicPetri
+using Catlab
+using Catlab.CategoricalAlgebra
+using Catalyst
+using OrdinaryDiffEq
+using DiffEqFlux
+using Plots
+using JSON
+using CSV
+using DataFrames
+
+#=
+THESE ARE STRATIFY FUNCTIONS FROM Structured-Epidemic-Modeling
+stratify(typed_model1, typed_model2) = ob(pullback(typed_model1, typed_model2))
+
+typed_stratify(typed_model1, typed_model2) =
+  compose(proj1(pullback(typed_model1, typed_model2)), typed_model1);
+=#
+
+abstract type AbstractStrataSpec end
+
+struct StrataSpec <: AbstractStrataSpec
+  tpn::ACSetTransformation
+  tlist::Vector{Vector{Symbol}}
+end
+
+"""Modify a typed petri net to add cross terms"""
+#=function add_cross_terms(pn_crossterms, type_system)
+
+  typed_pn, crossterms = deepcopy.(pn_crossterms)
+  pn = dom(typed_pn)
+  type_comps = Dict([k=>collect(v) for (k,v) in pairs(components(typed_pn))])
+  for (s_i,cts) in enumerate(crossterms)
+    for ct in cts 
+      type_ind = findfirst(==(ct), type_system[:tname])
+      is, os = [incident(type_system, type_ind, f) for f in [:it, :ot]]
+      new_t = add_part!(pn, :T; tname=ct)
+      add_parts!(pn, :I, length(is); is=s_i, it=new_t)
+      add_parts!(pn, :O, length(os); os=s_i, ot=new_t)
+      push!(type_comps[:T], type_ind)
+      append!(type_comps[:I], is); append!(type_comps[:O], os); 
+    end
+  end
+  return homomorphism(pn, codom(typed_pn); initial=type_comps, 
+                        type_components=(Name=x->nothing,),)
+end
+
+function add_cross_terms(ss::StrataSpec, type_system)
+  return add_cross_terms(ss.tpn=>ss.tlist, type_system)
+end=#
+
+function add_cross_terms!(ss::StrataSpec, type_system)
```

Comment:
This (and other functions) are in ASKEM.jl now; we should make sure to use those so that our demo shows the other performers how to use the functionality we are exposing. If there are functions here that are not in ASKEM.jl, they should be added to ASKEM.jl